src/main/scala/mesosphere/marathon/core/volume/VolumesModule.scala

```diff
@@ -0,0 +1,94 @@
+package mesosphere.marathon.core.volume
+
+import com.wix.accord._
+import mesosphere.marathon.core.volume.providers._
+import mesosphere.marathon.state._
+
+/**
+  * VolumeProvider is an interface implemented by storage volume providers
+  */
+trait VolumeProvider[+T <: Volume] {
```

Comment:
:+1: It is nice that you create an explicit interface to different pluggable VolumeProviders!